crates/src/event_subscriber.rs

```diff
@@ -205,6 +218,82 @@ impl LogEventStream {
     }
 }
 
+struct GrpcLogEventStream {
+    grpc_endpoint: String,
+    grpc_x_token: String,
+    sub_account: Pubkey,
+    event_tx: Sender<DriftEvent>,
+    commitment: CommitmentConfig,
+}
+
+impl GrpcLogEventStream {
+    /// Returns a future for running the configured log event stream
+    async fn stream_fn(self) {
+        let sub_account = self.sub_account;
+        info!(target: LOG_TARGET, "grpc log stream connecting: {sub_account:?}");
+
+        let mut grpc = DriftGrpcClient::new(self.grpc_endpoint.clone(), self.grpc_x_token.clone())
+            .grpc_connection_opts(GrpcConnectionOpts::default());
+
+        let (raw_event_tx, mut raw_event_rx): (
+            Sender<TransactionUpdate>,
+            Receiver<TransactionUpdate>,
+        ) = channel(256);
+
+        let raw_event_tx_clone = raw_event_tx.clone();
+        grpc.on_transaction(Box::new(move |tx_update: TransactionUpdate| {
+            raw_event_tx_clone.try_send(tx_update).unwrap();
```

Comment:
is it possible to handle the update inline here? maybe then can pass TransactionUpdate by ref

```suggestion
            let start = std::time::Instant::now();
            let slot = event.slot;
            self.process_log(event).await;
            let elapsed = start.elapsed();
            debug!(target: "grpc", "transaction slot: {}, len: {} callbacks took {:?}", slot, raw_event_rx.len(), elapsed);
```

Reply:
couldn't get this to work since `process_log` is async. Maybe i should change `grpc.on_transaction` to take async callbacks?